core/lib/contracts_loader/src/lib.rs

```diff
@@ -0,0 +1,135 @@
+use zksync_config::configs::contracts::{
+    chain::ChainContracts,
+    ecosystem::{EcosystemCommonContracts, L1SpecificContracts},
+    ChainSpecificContracts,
+};
+use zksync_contracts::{bridgehub_contract, state_transition_manager_contract};
+use zksync_eth_client::{CallFunctionArgs, EthInterface};
+use zksync_types::{
+    ethabi::{Contract, Token},
+    settlement::SettlementMode,
+    Address, L2ChainId,
+};
+
+/// Load contacts specific for each settlement layer, using bridgehub contract
+pub async fn load_settlement_layer_contracts(
+    sl_client: &dyn EthInterface,
+    bridgehub_address: Address,
+    l2_chain_id: L2ChainId,
+    multicall3: Option<Address>,
+) -> anyhow::Result<Option<ChainSpecificContracts>> {
+    let gateway_diamond_proxy: Address =
+        CallFunctionArgs::new("getZKChain", Token::Uint(l2_chain_id.as_u64().into()))
+            .for_contract(bridgehub_address, &bridgehub_contract())
+            .call(sl_client)
+            .await?;
+
+    if gateway_diamond_proxy.is_zero() {
+        return Ok(None);
+    }
+
+    let ctm_address =
+        CallFunctionArgs::new("chainTypeManager", Token::Uint(l2_chain_id.as_u64().into()))
+            .for_contract(bridgehub_address, &bridgehub_contract())
+            .call(sl_client)
+            .await?;
+
+    let server_notifier_addr = CallFunctionArgs::new("serverNotifierAddress", ())
```

Comment:
the same goes here, is it okay that it will return an error for the system before v27?